src/main.rs

```diff
@@ -1,34 +1,18 @@
-mod gccrs;
+use anyhow::{anyhow, Result};
+use cargo_gccrs::{Error, Gccrs, Wrapper};
 
-use gccrs::Gccrs;
-
-fn spawn_as_wrapper() {
-    // Skip `cargo` and `gccrs` in the invocation. Since we spawn a new cargo command,
-    // `cargo gccrs arg0 arg1` will become `cargo run arg0 arg1`
-    let mut cargo_gccrs = std::process::Command::new("cargo")
-        .env("RUSTC_WRAPPER", "cargo-gccrs")
-        .args(std::env::args().skip(2))
-        .spawn()
-        .expect("Unable to launch cargo-gccrs as RUSTC_WRAPPER");
-
-    cargo_gccrs
-        .wait()
-        .expect("Subprocess cargo-gccrs didn't complete properly");
-}
-
-fn main() {
+fn main() -> Result<()> {
     let args: Vec<String> = std::env::args().collect();
 
-    Gccrs::maybe_install().expect("gccrs should be installed");
+    Gccrs::maybe_install()?;
+
+    let first_arg = args.get(1).map(String::as_str);
 
-    let first_arg = args.get(1).expect("Invalid arguments");
+    let res = match first_arg {
+        Some("gccrs") => Wrapper::spawn(),
+        Some("rustc") => Gccrs::handle_rust_args(&args),
```

Comment:
Wait, does this handle both the `driver` and the `cargo` part? So if you call `cargo gccrs build`, this binary will execute itself with the same arguments?

When is this called with `gccrs` vs `rustc`?

Reply:
When you do `cargo gccrs`, cargo will run `cargo-gccrs gccrs`. When you do `RUSTC_WRAPPER=cargo-gccrs`, the spawned rustc processes will be `cargo-gccrs rustc`. (`RUSTC` instead of `RUSTC_WRAPPER` would omit the `rustc` part)